lib/ansible/utils/plugin_docs.py

```diff
@@ -342,15 +342,15 @@ def get_plugin_docs(plugin, plugin_type, loader, fragment_loader, verbose):
             newfile = _find_adjacent(filename, plugin, C.DOC_EXTENSIONS)
             if newfile:
                 docs = get_docstring(newfile, fragment_loader, verbose=verbose, collection_name=collection_name, plugin_type=plugin_type)
+                filename = newfile
         except Exception as e:
             raise AnsibleParserError('Adjacent file %s did not contain a DOCUMENTATION attribute (%s)' % (plugin, filename), orig_exc=e)
 
-    # got nothing, so this is 'undocumented', but lets populate at least some friendly info
-    if not docs[0]:
-        docs[0] = {'description': 'UNDOCUMENTED', 'short_description': 'UNDOCUMENTED'}
-
     # add extra data to docs[0] (aka 'DOCUMENTATION')
-    docs[0]['filename'] = filename
-    docs[0]['collection'] = collection_name
+    if docs[0] is None:
+        raise AnsibleParserError('No documentation availalbe for %s (%s)' % (plugin, filename))
```

Comment:
docs is a tuple, so item assignment fails due to the immutable type. Erroring here doesn't seem right since it will break external plugins which aren't necessarily expected to have documentation (and perhaps other items in the tuple are not None).

```diff
diff --git a/lib/ansible/utils/plugin_docs.py b/lib/ansible/utils/plugin_docs.py
index a4c1022992..304d25fe41 100644
--- a/lib/ansible/utils/plugin_docs.py
+++ b/lib/ansible/utils/plugin_docs.py
@@ -346,11 +346,14 @@ def get_plugin_docs(plugin, plugin_type, loader, fragment_loader, verbose):
         except Exception as e:
             raise AnsibleParserError('Adjacent file %s did not contain a DOCUMENTATION attribute (%s)' % (plugin, filename), orig_exc=e)
 
-    # add extra data to docs[0] (aka 'DOCUMENTATION')
+    # got nothing, so this is 'undocumented', but lets populate at least some friendly info
     if docs[0] is None:
-        raise AnsibleParserError('No documentation availalbe for %s (%s)' % (plugin, filename))
-    else:
-        docs[0]['filename'] = filename
-        docs[0]['collection'] = collection_name
+        undocumented = {'description': 'UNDOCUMENTED', 'short_description': 'UNDOCUMENTED'}
+        docs = docs[1:]
+        docs = (undocumented, *docs)
+
+    # add extra data to docs[0] (aka 'DOCUMENTATION')
+    docs[0]['filename'] = filename
+    docs[0]['collection'] = collection_name
```

Reply:
split doesn't have documentation so it can be used to see the issue:
```
$ ansible-doc ansible.builtin.split -t filter -vvv
...
Traceback (most recent call last):
  File "/home/liveuser/ansible/bin/ansible-doc", line 667, in _get_plugins_docs
    doc, plainexamples, returndocs, metadata = get_plugin_docs(plugin, plugin_type, loader, fragment_loader, (context.CLIARGS['verbosity'] > 0))
  File "/home/liveuser/ansible/lib/ansible/utils/plugin_docs.py", line 351, in get_plugin_docs
    raise AnsibleParserError('No documentation availalbe for %s (%s)' % (plugin, filename))
ansible.errors.AnsibleParserError: No documentation availalbe for ansible.builtin.split (/home/liveuser/ansible/lib/ansible/plugins/filter/core.py)
ERROR! filter ansible.builtin.split missing documentation (or could not parse documentation): No documentation availalbe for ansible.builtin.split (/home/liveuser/ansible/lib/ansible/plugins/filter/core.py)
```

Reply:
the error here is expected at the caller, which handles the 'undocumented' already, i restored it since it created issues for the docs build

Reply:
I'm not sure, since missing docs is still an error for other plugins in 2.13 (cannot test filters/tests since not available there, just removed DOCUMENTATION from ssh connection plugin).
```
ansible-doc [core 2.13.1.post0] (stable-2.13 99217ca2b6) last updated 2022/06/27 16:32:35 (GMT -400)
  config file = None
  configured module search path = ['/home/bcoca/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/bcoca/work/ansible/lib/ansible
  ansible collection location = /home/bcoca/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/bcoca/work/ansible/bin/ansible-doc
  python version = 3.9.13 (main, Sep  1 2022, 19:27:49) [GCC 11.3.0]
  jinja version = 3.1.2
  libyaml = True
No config file found; using defaults
Traceback (most recent call last):
  File "/home/bcoca/work/ansible/bin/ansible-doc", line 667, in _get_plugins_docs
    doc, plainexamples, returndocs, metadata = DocCLI._get_plugin_doc(plugin, plugin_type, loader, search_paths)
  File "/home/bcoca/work/ansible/bin/ansible-doc", line 877, in _get_plugin_doc
    raise ValueError('%s did not contain a DOCUMENTATION attribute' % plugin)
ValueError: ansible.builtin.ssh did not contain a DOCUMENTATION attribute
ERROR! connection ansible.builtin.ssh missing documentation (or could not parse documentation): ansible.builtin.ssh did not contain a DOCUMENTATION attribute
```
I did change the type of error though, is that the issue?

Reply:
Okay, that makes sense. I forgot -t filter didn't work on 2.13.